prclz/etl/split_bldgs.py

```diff
@@ -0,0 +1,104 @@
+import argparse
+from typing import List, Union
+from logging import basicConfig, info
+from pathlib import Path 
+
+import geopandas as gpd
+import pandas as pd
+
+def gadm_dir_to_path(gadm_dir: str) -> str:
+	"""
+	For a given country,the GADM dir contains multiple file levels
+	so this convenience function just returns the path to the highest
+	resolution gadm file within the directory, which changes from 
+	country to country
+
+	Args:
+		gadm_dir: directory containing all gadm files for a country
+	
+	Returns:
+		Path to specific gadm file
+	"""
+	sort_fn = lambda p: int(p.stem.split("_")[-1])
+	gadm_dir = Path(gadm_dir)
+	files = [p for p in gadm_dir.iterdir() if ".shp" in p.name]
+	files.sort(key=sort_fn)
+	return files[-1]
+
+
+def clean_gadm_cols(gadms: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+	"""
+	The GADM files have columns GID_0, GID_1, ... , GID_n depending
+	on the gadm resolution. So this extracts the highest res gadms
+	and standardizes the col names
+	"""
+	sort_fn = lambda n: int(n.replace("GID_",""))
+	cols = [c for c in gadms.columns if "GID_" in c]
+	cols.sort(key=sort_fn)
+
+	cc = cols[0]
+	gadm = cols[-1]
+	std_gadms = gadms[[cc, gadm, 'geometry']].copy()
```

Comment:
is there a reason to `copy`?

consider: 
```python
cc, *_, gadm = sorted([c for c in gadms.columns if "GID_" in c], key = lambda n: int(n.replace("GID_",""))
return gadms[[cc, gadm, 'geometry']].rename(columns={cc: 'gadm_code', gadm: 'gadm'})
```

Reply:
Good point, dropped the copy.
I don't see the point of jamming everything into one line. There's less than 10 things in the list and the function isn't a performance bottleneck so this just seems way less legible for no real increase in speed. Or am i missing something?